source/common/router/config_impl.cc

```diff
@@ -1004,10 +1016,27 @@ VirtualHostImpl::VirtualHostImpl(const envoy::api::v2::route::VirtualHost& virtu
 
 VirtualHostImpl::VirtualClusterEntry::VirtualClusterEntry(
     const envoy::api::v2::route::VirtualCluster& virtual_cluster, Stats::StatNamePool& pool)
-    : pattern_(RegexUtil::parseRegex(virtual_cluster.pattern())),
-      stat_name_(pool.add(virtual_cluster.name())) {
+    : stat_name_(pool.add(virtual_cluster.name())) {
+  if (virtual_cluster.pattern().empty() == virtual_cluster.headers().empty()) {
+    throw EnvoyException("virtual clusters must define either 'pattern' or 'headers'");
+  }
+
+  if (!virtual_cluster.pattern().empty()) {
+    envoy::api::v2::route::HeaderMatcher matcher_config;
+    matcher_config.set_name(Http::Headers::get().Path.get());
+    matcher_config.set_regex_match(virtual_cluster.pattern());
```

Comment:
BTW, did you make any progress on some structural or check_format approach to banning or containing the old regex use?

Reply:
It's pushed already...

Reply:
For my own education, which PR?

Reply:
This one :) https://github.com/envoyproxy/envoy/pull/7878/files#diff-2e4b5720c93b5c27bb0c8c7c4312aaa8R66